vllm/utils.py

```diff
@@ -222,11 +222,16 @@ async def producer(i: int, iterator: AsyncIterator[T]):
     ]
 
     async def consumer():
-        while not all(finished) or not queue.empty():
-            item = await queue.get()
-            if isinstance(item, Exception):
-                raise item
-            yield item
+        try:
+            while not all(finished) or not queue.empty():
+                item = await queue.get()
+                if isinstance(item, Exception):
+                    raise item
+                yield item
+        except (Exception, asyncio.CancelledError) as e:
+            for task in _tasks:
+                task.cancel(str(e))  # type: ignore
```

Comment:
[`asyncio.Future.cancel()`](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.cancel) only accepts `msg` parameter in Python 3.9 onwards. Since vLLM currently supports Python 3.8, we cannot pass the error string yet.

Reply:
Thanks for pointing out that